valid-palindrome/changchanghwang.go

```diff
@@ -0,0 +1,37 @@
+// 풀이
+// alphabet, number만 걸러내서 소문자로 바꾼 후 reverse한 문자열, 걸러낸 문자열과 비교하여 같은지 확인한다.
+
+// O(n) time complexity
+// O(n) space complexity
+func isPalindrome(s string) bool {
+	regex, _ := regexp.Compile("[a-zA-Z0-9]")
+	reverseAndFilteredString := ""
+	filteredString := ""
+
+	for _, char := range s {
+		if regex.MatchString(string(char)) {
+			c := strings.ToLower(string(char))
```

Comment:
```go
c := strings.ToLower(string(char))
```
`char` 하나씩 `string()`함수로 타입 변환하여 `strings.ToLower()`함수를 적용하는 것보다는, 입력으로 주어진 전체 문자열 `s`를 통째로 `strings.ToLower()`를 적용시키는 것이 더 깔끔한 코드가 될 것 같습니다

```suggestion
        s = strings.ToLower(s)
	regex, _ := regexp.Compile("[a-zA-Z0-9]")
	reverseAndFilteredString := ""
	filteredString := ""

	for _, char := range s {
		if regex.MatchString(string(char)) {
```

Reply:
앗 아무생각없이 짰더니 저런 안좋은 코드가 나왔네요.. ㅎㅎ 수정했습니다 감사합니다!

Reply:
아이구 안 좋은 코드는 아니에요!!!